### PR TITLE
Update grab bag

### PR DESCRIPTION
**Pull request by workingjubilee**
This PR slides a bunch of crate versions forward until suddenly a bunch of deps fall out of the tree!
In doing so this mostly picks up a version bump in the `redox_users` crate which makes most of the features default to optional.

crossbeam-utils 0.7 => 0.8.3 (where applicable)
https://github.com/crossbeam-rs/crossbeam/blob/master/crossbeam-utils/CHANGELOG.md
directories 3.0.1 => 3.0.2
ignore 0.4.16 => 0.4.17
tempfile 3.0.5 => tempfile 3.2

Removes constant_time_eq from deps exceptions
Removes arrayref from deps exceptions
And also removes:
- blake2b_simd
- const_fn (the package, not the feature)
- constant_time_eq
- redox_users 0.3.4
- rust-argon2

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
Generally it's best to split these up in case there's any regressions, and provide links to bumped dependency changelogs if possible. But let's try a perf run to start:

@bors try @rust-timer queue

**Comment by workingjubilee** [comment]
Alas, [crossbeam has a changelog](https://github.com/crossbeam-rs/crossbeam/blob/master/crossbeam-utils/CHANGELOG.md) but none of the others do and instead I went crawling individual commits to find this out! :D
e.g. https://github.com/dirs-dev/dirs-sys-rs/commit/e776c08a220bdb38cc64b29d7912fdee42143343

**Comment by workingjubilee** [comment]
Huh, that's weird, I thought I rebased that already. Sorry!

**Comment by workingjubilee** [comment]
My latest commit assumes the instructions given by the logbot are correct and therefore adds rand_core 0.5.1 to rustc-workspace-hack. Apologies if it is not, I have not encountered this flavor of error before and I am not entirely sure of how to proceed.

**Comment by workingjubilee** [comment]
Nothing dramatic, but almost no red and all green. 💚

**Comment by workingjubilee** [comment]
> Generally it's best to split these up in case there's any regressions, and provide links to bumped dependency changelogs if possible.

I could probably have left the tempfile change out, it's true.

The rest kind of seemed like it "goes together" because of the motion towards crossbeam-utils 0.8.x (and the bugfix in crossbeam-utils 0.8.2), and I discovered the sudden "dep cascade" only when poking around after the tempfile tweak. But I don't really understand the resolver from the user point of view. ^^;

File: src/tools/tidy/src/deps.rs [hunk 1]
```diff
@@ -35,11 +35,9 @@ const EXCEPTIONS: &[(&str, &str)] = &[
     ("ryu", "Apache-2.0 OR BSL-1.0"),   // rls/cargo/... (because of serde)
     ("bytesize", "Apache-2.0"),         // cargo
     ("im-rc", "MPL-2.0+"),              // cargo
-    ("constant_time_eq", "CC0-1.0"),    // rustfmt
     ("sized-chunks", "MPL-2.0+"),       // cargo via im-rc
     ("bitmaps", "MPL-2.0+"),            // cargo via im-rc
     ("crossbeam-queue", "MIT/Apache-2.0 AND BSD-2-Clause"), // rls via rayon
-    ("arrayref", "BSD-2-Clause"),       // cargo-miri/directories/.../rust-argon2 (redox)
```
Comment (bjorn3) [inline src/tools/tidy/src/deps.rs:42]: Could you also remove unused entries from `PERMITTED_DEPENDENCIES`?

Comment (workingjubilee) [reply src/tools/tidy/src/deps.rs:42]: Erm, which ones? These are transitive dependencies, and don't seem to be covered by that list.

Comment (bjorn3) [reply src/tools/tidy/src/deps.rs:42]: I thought it also required listing all transitive dependencies. This indeed seems to not be the case.

**Pull request merged**